src/pytorch_lightning/core/saving.py

```diff
@@ -171,15 +170,15 @@ def on_hpc_load(self, checkpoint: Dict[str, Any]) -> None:
 
 
 def _load_from_checkpoint(
-    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
+    cls: Union[Type["ModelIO"], Type["pl.LightningModule"], Type["pl.LightningDataModule"]],
     checkpoint_path: Union[str, IO],
-    map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
+    map_location: _MAP_LOCATION_TYPES = None,
     hparams_file: Optional[str] = None,
-    strict: Optional[bool] = None,
+    strict: bool = True,
     **kwargs: Any,
-) -> Any:
+) -> Union["pl.LightningModule", "pl.LightningDataModule"]:
     if map_location is None:
-        map_location = lambda storage, loc: storage
+        map_location = cast(_MAP_LOCATION_TYPES, lambda storage, loc: storage)
```

Comment:
```suggestion
        map_location = cast(_MAP_LOCATION_TYPE, lambda storage, loc: storage)
```
just `_MAP_LOCATION_TYPE`

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/148bb26d4275a0b263d036f6d15e57e384ed94a5